terraform/data.tf

```
@@ -0,0 +1,30 @@
+data "aws_vpc" "vpc" {
+  tags = { Name = "trade_tariff_${var.environment}_vpc" }
+}
+
+data "aws_subnets" "private" {
+  filter {
+    name   = "vpc-id"
+    values = [data.aws_vpc.vpc.id]
+  }
+
+  tags = {
+    Name = "*private*"
+  }
+}
+
+data "aws_lb_target_group" "this" {
+  name = "trade-tariff-be-tg-${var.environment}"
+}
+
+data "aws_security_group" "this" {
+  name = "trade-tariff-alb-security-group-${var.environment}"
```

```suggestion
  name = "trade-tariff-ecs-security-group-${var.environment}"
```
We'll want to use the ECS security group here, this is the one I have fixed to allow egress out to AWS services so that the task can pull the containers and access secrets